crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -212,9 +212,32 @@ impl<'block> BrilligBlock<'block> {
                 );
                 self.convert_ssa_binary(binary, dfg, result_register);
             }
-            Instruction::Constrain(value) => {
-                let condition = self.convert_ssa_register_value(*value, dfg);
-                self.brillig_context.constrain_instruction(condition);
+            Instruction::Constrain(lhs, rhs) => {
+                // If either value is a constant value of 1 then we can apply the constrain instruction to the other value directly.
+                match (dfg.get_numeric_constant(*lhs), dfg.get_numeric_constant(*rhs)) {
+                    (Some(lhs_constant), _) if lhs_constant == FieldElement::one() => {
+                        let condition = self.convert_ssa_register_value(*rhs, dfg);
+
+                        self.brillig_context.constrain_instruction(condition);
+                    }
+                    (_, Some(rhs_constant)) if rhs_constant == FieldElement::one() => {
+                        let condition = self.convert_ssa_register_value(*lhs, dfg);
+
+                        self.brillig_context.constrain_instruction(condition);
+                    }
+                    (_, _) => {
+                        let condition = self.brillig_context.allocate_register();
+
+                        self.convert_ssa_binary(
+                            &Binary { lhs: *lhs, rhs: *rhs, operator: BinaryOp::Eq },
+                            dfg,
+                            condition,
+                        );
+
+                        self.brillig_context.constrain_instruction(condition);
+                        self.brillig_context.deallocate_register(condition);
+                    }
+                }
```

Comment:
This is intended to address that brillig doesn't perform the `constrain(x == true) => constrain(x)` optimisation. We should be able to remove this once #2066 is addressed.

Reply:
I thought you might say this but I was enjoying how much this PR reduced the bytecode size. Will revert this.